### PR TITLE
converted dataplane actions to packet out/in for loopback

### PR DESCRIPTION
**Pull request by abhilashendurthi**

### CODE REVIEW

File: pkg/framework/dataplane/loopback.go [hunk 1]
```diff
@@ -108,7 +115,12 @@ func (ldp *loopbackDataPlane) verifyOnPort(port uint32, pkts [][]byte) bool {
 			return result
 		// The default case is only for passing CI and will be updated/removed once implemetation completes
 		default:
-			time.Sleep(1 * time.Second)
+			if len(pkts) > 0 {
+				pi := convertToPktIn(port, pkts[0])
```
[the remark this answers is not in the document]
Comment (abhilashendurthi) [reply pkg/framework/dataplane/loopback.go]: yes, thanks for catching that. The correct functionality would be looping for all the packets on that port and returning true on receiving them. Let me know if my understanding is wrong.

[the remark this answers is not in the document]
Comment (yoooou) [reply pkg/framework/dataplane/loopback.go]: Need more discussion on how to handle multiple traffic expectations on different ports - we shouldn't assume that the order of egressed packets (on different ports) is deterministic. E.g. we shouldn't be blocked when we process traffic expectation on port-1 but receive a packet on port-2 first.

File: pkg/framework/dataplane/loopback.go [hunk 2]
```diff
@@ -108,7 +115,18 @@ func (ldp *loopbackDataPlane) verifyOnPort(port uint32, pkts [][]byte) bool {
 			return result
 		// The default case is only for passing CI and will be updated/removed once implemetation completes
 		default:
```
Comment (yoooou) [inline pkg/framework/dataplane/loopback.go]: In this case it always runs the default branch and the timer won't work which makes it inconsistent with the direct mode

Comment (yoooou) [reply pkg/framework/dataplane/loopback.go]: We'll keep as is in this patch and address the issue after more discussions

File: pkg/framework/dataplane/loopback.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (abhilashendurthi) [reply pkg/framework/dataplane/loopback.go]: Made changes to handle multiple traffic expectations by sorting PacketIns into multiple channels based on out ports

File: pkg/framework/dataplane/loopback.go (hunk 2 above)
Comment (abhilashendurthi) [reply pkg/framework/dataplane/loopback.go]: Keeping as is in this patch, to be fixed after more discussions.

**Pull request merged**